numpy/lib/shape_base.py

```diff
@@ -134,9 +134,9 @@ def apply_along_axis(func1d, axis, arr, *args, **kwargs):
         buff = res.__array_prepare__(buff)
 
     # save the first result, then compute and save all remaining results
-    buff[ind0] = res
+    buff[ind0] = res[()]
```

Comment:
Ok, comment added. I think that a helper function would be more widely applicable, so belongs in a separate MAINT PR. I've created #8681 to discuss what it should be named